src/target/move.c

```diff
@@ -353,48 +328,52 @@ void parametric_move_front(float distance, float end_linear_speed)
 /**
  * @brief Move left or right into the next cell.
  *
- * @param[in] step_direction Turn direction (left or right).
+ * @param[in] movement Turn direction (left or right).
+ * @param[in] speed The speed level at which to move side.
  */
-void move_side(enum step_direction side)
+void move_side(enum movement turn, uint8_t speed)
 {
 	enable_walls_control();
 	target_straight(current_cell_start_micrometers,
-			CELL_DIMENSION / 2 - get_turn_radius(),
-			get_turn_linear_speed());
+			get_move_turn_space(turn, speed),
+			get_move_turn_linear_speed(turn, speed));
 	disable_walls_control();
-	turn_side(side);
+	speed_turn(turn, speed);
 	enable_walls_control();
 	target_straight(get_encoder_average_micrometers(),
-			CELL_DIMENSION / 2 - get_turn_radius(),
+			get_move_turn_space(turn, speed),
 			get_max_end_linear_speed());
 	entered_next_cell();
 }
 
 /**
  * @brief Move back into the previous cell.
+ *
+ * @param[in] speed The speed level at which to move back.
  */
-void move_back(void)
+void move_back(uint8_t speed)
 {
 	stop_middle();
-	turn_back();
+	turn_back(speed);
 	move_front();
 }
 
 /**
  * @brief Move into the next cell according to a movement direction.
  *
  * @param[in] direction Movement direction.
+ * @param[in] speed The speed level at which to move to the next cell.
  */
-void move(enum step_direction direction)
+void move(enum step_direction direction, uint8_t speed)
 {
 	if (direction == LEFT)
-		move_side(LEFT);
+		move_side(MOVE_LEFT, speed);
 	else if (direction == RIGHT)
-		move_side(RIGHT);
+		move_side(MOVE_RIGHT, speed);
 	else if (direction == FRONT)
 		move_front();
```

Comment:
Why it does not include speed dependency?

Reply:
Not implemented yet... :sweat_smile:

Reply:
ahh, right! :+1: